src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,33 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+
+import { getUser } from '../../api';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
 import { Loader } from '../Loader';
 
-export const TodoModal: React.FC = () => {
+interface Props {
+  selectedTodo: Todo;
+  setSelectedTodo: (todo: Todo | null) => void;
```

Comment:
Your modal never sets Todo, it only clears it

I would suggest:
```suggestion
  onClose: () => void;
```